perun-core/src/main/java/cz/metacentrum/perun/core/impl/PerunRolesLoader.java

```diff
@@ -71,36 +77,22 @@ public void loadPerunRoles(JdbcPerunTemplate jdbc) {
 	 *
 	 * @return list of PerunPolicies
 	 */
-	public List<PerunPolicy> loadPerunPolicies() {
-		List<PerunPolicy> policies = new ArrayList<>();
+	public Set<PerunPolicy> loadPerunPolicies() {
+		Set<PerunPolicy> policies = new HashSet<>();
 
 		try {
-			JsonNode rootNode = loadConfigurationFile();
-			//Fetch all policies from the configuration file
-			JsonNode policiesNode = rootNode.get("perun_policies");
-
-			// For each policy node construct PerunPolicy and add it to the list
-			Iterator<String> policyNames = policiesNode.fieldNames();
-			while(policyNames.hasNext()) {
-				String policyName = policyNames.next();
-				JsonNode policyNode = policiesNode.get(policyName);
-				List<Map<String, String>> perunRoles = new ArrayList<>();
-				JsonNode perunRolesNode = policyNode.get("policy_roles");
-
-				//Field policy_roles is saved as List of maps in the for loop
-				for (JsonNode perunRoleNode : perunRolesNode) {
-					Map<String, String> innerRoleMap = createmapFromJsonNode(perunRoleNode);
-					perunRoles.add(innerRoleMap);
-				}
-
-				//Field include_policies is saved as List of Strings.
-				List<String> includePolicies = new ArrayList<>(objectMapper.convertValue(policyNode.get("include_policies"), new TypeReference<List<String>>() {
-				}));
-
-				policies.add(new PerunPolicy(policyName, perunRoles, includePolicies));
+			JsonNode rootNode;
+			if (secondaryConfigurationPath != null) {
+				rootNode = loadConfigurationFile(secondaryConfigurationPath);
+				policies.addAll(loadPoliciesFromJsonNode(rootNode));
 			}
+			rootNode = loadConfigurationFile(configurationPath);
```

Comment:
Shouldn't we first load normal configuration and then overwrite it with secondary config (as we do in ```loadPerunRolesManagement()```)?

Reply:
Ok, @balcirakpeter corrected me, that its on purpose, because of behavior of adding policies method.